worker/uniter/runner/context/secrets.go

```diff
@@ -27,7 +28,44 @@ func newSecretsChangeRecorder(logger loggo.Logger) *secretsChangeRecorder {
 	}
 }
 
-func (s *secretsChangeRecorder) update(arg uniter.SecretUpdateArg) {
+func (s *secretsChangeRecorder) create(arg uniter.SecretCreateArg) {
+	for i, d := range s.pendingDeletes {
+		if d.ID == arg.URI.ID {
+			s.pendingDeletes = append(s.pendingDeletes[:i], s.pendingDeletes[i+1:]...)
+			break
+		}
+	}
+	s.pendingCreates = append(s.pendingCreates, arg)
+}
+
+func (s *secretsChangeRecorder) update(arg uniter.SecretUpsertArg) {
+	for i, d := range s.pendingDeletes {
+		if d.ID == arg.URI.ID {
+			s.pendingDeletes = append(s.pendingDeletes[:i], s.pendingDeletes[i+1:]...)
+			break
+		}
+	}
+	for i, c := range s.pendingCreates {
+		if c.URI.ID == arg.URI.ID {
```

Comment:
Sometimes worth inverting the expression so we can avoid nested branch statements.

Reply:
fixed